cmd/reset/reset-consumer-group-offset.go

```diff
@@ -29,9 +29,10 @@ func newResetOffsetCmd() *cobra.Command {
 
 	cmdResetOffset.Flags().BoolVarP(&offsetFlags.OldestOffset, "oldest", "", false, "set the offset to oldest offset (for all partitions or the specified partition)")
 	cmdResetOffset.Flags().BoolVarP(&offsetFlags.NewestOffset, "newest", "", false, "set the offset to newest offset (for all partitions or the specified partition)")
+	cmdResetOffset.Flags().BoolVarP(&offsetFlags.AllTopics, "all-topics", "", false, "do the operation for all topics in the consumer group")
 	cmdResetOffset.Flags().Int64VarP(&offsetFlags.Offset, "offset", "", -1, "set offset to this value. offset with value -1 is ignored")
 	cmdResetOffset.Flags().Int32VarP(&offsetFlags.Partition, "partition", "p", -1, "partition to apply the offset. -1 stands for all partitions")
-	cmdResetOffset.Flags().StringVarP(&offsetFlags.Topic, "topic", "t", offsetFlags.Topic, "topic to change offset for")
+	cmdResetOffset.Flags().StringArrayVarP(&offsetFlags.Topic, "topic", "t", offsetFlags.Topic, "one ore more topics to change offset for")
```

Comment:
I changed the semantics of `AllTopics` so that not all topics in the cluster are taken into account, but only all topics that are already part of the consumer-group.

Additionally, I removed the parameter to read topiclist from file, but instead added the possibility to provide multiple topics via `topic` parameter.
I think that gives enough flexibility and keeps the parameters a bit "cleaner". For scenarios where one really wants to do this for topics in a file, it is reasonable to write a small script around it.

Reply:
We actually have the use case here for a list of topics read from a file. 

We have an app which parse a topic name using the regex parser into a list of topics which is more than 400 - just imagine if you supply --topic - you just can not lol. 

--all-topics seems to be a bit overkill but hey, the kafka shell tools allow that options too, thus the first thing I can think of is to pick up what they support.

like this

```
kafka-consumer-groups.sh --bootstrap-server kafka-host:9092 --group my-group 
--reset-offsets --to-earliest --all-topics --execute
```

You can do a shell for loop and call kafkactl per topics but it would be still slow. If you actually want shell to spawn multiple kafkactl to change, then why just do it in the tool more efficient.

Personally I wish to keep --all-topics and --topic-file-list and your change to allow multiple topics. 

The use case is that we change the config of the app to use new consumer group, and we do not want it re-poll the history to overload the extractor and loader.

Reply:
honestly, this seems to be a rather exotic usecase.

> You can do a shell for loop and call kafkactl per topics but it would be still slow
How about iterating through your topic list in a script and call kafkactl with bulks of ~10 topics to speed it up?

Reply:
if you use bash u need to do something like
```
count=1
while [ $count -lt 11 ]; do
# run cmd with & at at end
if # well give up here :) 
```
yuk now, I need to google search to find out how to get the command back and manage what golang errorgroup does similar things.

So I will write a golang wrapper to start a shell to do so which is much easier to go beyond bash provides. Or pythons script

Reply:
in your case its probably easier to write the complete thing in python. Perhaps this here helps: https://github.com/confluentinc/confluent-kafka-python/issues/201#issuecomment-308856631

Reply:
Thanks for the idea. 
I can use golang wrapper - better managed, in here we use golang for everything even simple automation adhoc tasks etc ... so it is just easier to wrap it around. Best is some remote environment ther eis no python, but wrap golang in can run everywhere, nearly ...